### PR TITLE
Improve expired auth and id token handling

### PR DESCRIPTION
**Pull request by znicholasbrown**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
Improves handling of expired id and authorization tokens on first load, and adds a bunch more tests around the `login` method. Also adds routing to the help page when a user doesn't meet auth requirements but the app loads, instead of routing to an empty welcome screen.

### CODE REVIEW

**Review by zhen0 — COMMENTED** [review]
One suggestion about checking for tenant.  Also, I think we should add the access-denied route to the list of pages where the navBar is hidden.  (I think it's the showNav computed property?).  It looked like this for me: 

Screen Shot 2021-06-17 at 9 56 55 AM

File: src/App.vue [hunk 1]
```diff
@@ -359,18 +359,32 @@ export default {
...
+    if (this.isCloud) {
+      if (
+        window.location.pathname?.includes('logout') ||
+        window.location.pathname?.includes('access-denied')
+      )
+        return
+
+      // If the application has loaded but the user isn't authenticated, authorized, or has no tenant
+      // redirect them to the help screen
+      if (!this.isAuthorized || !this.isAuthenticated || !this.tenant) {
```
[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/App.vue:371]: Ah yeah good point

**Comment by znicholasbrown** [comment]
Chatted with @zhen0 about that screen and we're going to leave the route as-is so that users can still log out even when they're in a bad state like the user in the screenshot above. For users who fail the authorization process, the nav bar will be significantly truncated but will still allow logging out

**Review by zhen0 — APPROVED** [review]
Thansk @znicholasbrown LGTM!

**Pull request merged**